ui/apps/platform/src/Containers/Policies/Wizard/Step1/MitreTacticSelect.tsx

```diff
@@ -26,44 +34,70 @@ function MitreTacticSelect({
     mitreAttackVectors,
     tacticId,
 }: MitreTacticSelectProps): ReactElement {
-    const [isOpen, setIsOpen] = useState(false);
+    const { isOpen, setIsOpen, onSelect, onToggle } = useSelectState(handleSelectOption);
 
-    function onSelect(_event, optionId) {
-        setIsOpen(false);
-        handleSelectOption(optionId);
-    }
+    // Find the display content for the selected value
+    const getDisplayContent = (): React.ReactNode => {
```

Comment:
+1 to my previous comment. This would be better either as a Component, or rendered directly via ternaries.